src/dirent_lookup.h

```diff
@@ -53,29 +53,28 @@ class DirentLookup
   typedef std::map<char, entry_index_t> NamespaceBoundaryCache;
 
 private: // data
-  const TDirentAccessor* direntAccessor = nullptr;
+  const TDirentAccessor& direntAccessor;
+  const entry_index_type direntCount;
 
   mutable NamespaceBoundaryCache namespaceBoundaryCache;
   mutable std::mutex cacheAccessMutex;
 
-  entry_index_type direntCount = 0;
   NarrowDown lookupGrid;
 };
 
 template<class TDirentAccessor>
 std::string
 DirentLookup<TDirentAccessor>::getDirentKey(entry_index_type i) const
 {
-  const auto d = direntAccessor->getDirent(entry_index_t(i));
+  const auto d = direntAccessor.getDirent(entry_index_t(i));
   return d->getNamespace() + d->getUrl();
 }
 
 template<class TDirentAccessor>
 DirentLookup<TDirentAccessor>::DirentLookup(const TDirentAccessor* _direntAccessor, entry_index_type cacheEntryCount)
+  : direntAccessor(*_direntAccessor)
+  , direntCount(direntAccessor.getDirentCount())
 {
-  ASSERT(direntAccessor == nullptr, ==, true);
```

Comment:
This remove the ASSERT.
I wonder if we should not change the signature of the constructor itself to accept a reference instead of a pointer. So we would not need the ASSERT.

Reply:
The assert was meaningless since it checked the data member (rather than the parameter). Obviously it was inherited from older code when this function was not a constructor. We may need an opposite assertion that checks the parameter. However the need for such an assertion is much less justified.

> I wonder if we should not change the signature of the constructor itself to accept a reference instead of a pointer. So we would not need the ASSERT.

I am strongly against passing parameters by references into functions that internally store addresses of the referred objects. A pointer type better communicates the idea (at the location of the call) that the address of the object will be stored and therefore the object must stay alive after the call.

Reply:
> I am strongly against passing parameters by references into functions that internally store addresses of the referred objects. A pointer type better communicates the idea (at the location of the call) that the address of the object will be stored and therefore the object must stay alive after the call.

You are totally right. I was a bit hurry in my proposal.